src/jank/clojure/core.jank

```diff
@@ -512,6 +512,27 @@
           (recur ret (first ks) (next ks))
           ret)))))
 
+(def pop!)
```

Comment:
@jeaye, what's the purpose of `(def pop!)` before an actual `(def pop! ...)`?

Reply:
That's only necessary to first declare `pop!`, in the case where it's used recursively. Please use `defn` for all of these fns, not `def` and `fn*`. Also, there's no need for the declaration ahead of time. `defn` does that automatically.

Also, please use `let`, rather than `let*`.